src/Control/Middleware/CanonicalURLMiddleware.php

```diff
@@ -76,6 +76,32 @@ class CanonicalURLMiddleware implements HTTPMiddleware
      */
     protected $forceSSLDomain = null;
 
+    /**
+     * Set if we should force trailing slash redirection
+     *
+     * @var bool
+     */
+    protected $forceTrailingSlash = true;
```

Comment:
While adding tests for this I found that having this as config doesn't work as expected - so I've reverted this back to being a protected instance variable.

Reply:
What if the config is on 'Director' or 'Controller'?

Reply:
What do you mean?

Reply:
I mean what if the config is set on `Controller` and retrieved from there wherever it's needed? 

like `Controller::config()->get('force_trailing_slash)`.

`add_trailing_slash` is also configured on Controller, so it would kind of make sense to have both in the same spot?

Reply:
`forceTrailingSlash` is only for the redirect isn't it? So only the middleware ever needs that info. Whereas `add_trailing_slash` is for whether there's a trailing slash on the end of URLs generated by framework or not.

Reply:
correct.